### ISSUE TITLE
[hardware] --critical-count-... --no-component proper detection

### ISSUE THREAD

**Issue by UrBnW** [centreon/centreon-plugins#2062]
Hi,

Let's take this example where a hardware component (out of 2 selected) is not detected :
```
--plugin=hardware::server::dell::idrac::snmp::plugin --mode=hardware --component='^(coolingunit|coolingdevice)$'
OK: All 8 components are ok [8/8 cooling devices].
```

With `--critical-count-coolingdevice=10:`, it returns :
`CRITICAL: '8' components 'coolingdevice' checked`

With `--critical-count-coolingunit=1:`, it returns :
`OK: All 8 components are ok [8/8 cooling devices].`

With `--no-component`, it returns :
`OK: All 8 components are ok [8/8 cooling devices].`

In other words, when several components are selected, there's no way to get an alert if one of these components is missing.
I then think `--warning-count-` / `--critical-count-` should be proceeded before ignoring the component.
I even wonder whether or not `--no-component` should be applied per-component, rather than globally.

Thx 👍

**Comment by garnier-quentin** [comment]
You set the threshold on the number you should have. I don't understand.

**Comment by garnier-quentin** [comment]
You want an alert if one of the component is missing (so it's 9 or 8). The first threshold is ok.

**Comment by UrBnW** [comment]
No, here I want an alert if `coolingunit` is not detected.
And it does not work, whatever the option I use, `--critical-count-coolingunit=1:` or `-no-component`.
My first example shows you that `--critical-count` only works if the component exists (here with `coolingdevice`).

**Comment by garnier-quentin** [comment]
i understand. you want to get the perfdata even if it's 0 for components.

**Comment by UrBnW** [comment]
Yes, so that one who monitors a new device using a defined template will get an alert saying :
"_hey, I'm asked to monitor `coolingunit`, but there's no `coolingunit` on your device !_"
And then he can take the required actions.

As said above, I even wonder whether or not `--no-component` should be applied per-component (so that here `coolingunit` would trigger), rather than globally.

**Comment by garnier-quentin** [comment]
i won't change --no-component. I can add an option to check component with 0 count.

**Comment by UrBnW** [comment]
In this case, if `--critical-count-coolingunit` works, it's OK for me :+1:

**Comment by garnier-quentin** [comment]
Yes it will.

**Comment by garnier-quentin** [comment]
You can test with option: --no-component-count

**Comment by UrBnW** [comment]
Works perfectly, many thanks @garnier-quentin 👍

```
--critical-count-coolingdevice=1: --critical-count-coolingunit=1: --no-component-count
CRITICAL: '0' components 'coolingunit' checked |
```

```
--critical-count-coolingdevice=1: --critical-count-coolingunit=0 --no-component-count
OK: All 8 components are ok [8/8 cooling devices, 0/0 cooling units]. |
```
